### DISCUSSION TITLE
Looking for maintainers for `psbt-v0` crate

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#3404] [General]
We now have a new crate https://github.com/rust-bitcoin/rust-psbt-v0 and are looking for devs to put their hand up to help maintain it. 

Intentions for the crate are (these are my personal thoughts, needs ack from @apoelstra):

1. One ack policy (one ack from maintainer required before merge)
2. Use merge script to merge PRs (requires (1))
3. Move faster than `rust-bitcoin`

Current released version is `0.1.0` and is an exact import of the `psbt` module from `rust-bitcoin 0.32.2`. As such it is a drop in replacement and is interoperable with current release of `bitcoin`  and `miniscript` by serializing to/from `Psbt` or manually converting since all fields are public.

Roadmap is up for discussion, we are looking for maintainers with a vested interest in the crate i.e., active users of the crate that want to see it progress. I am not an active user which is a weakness as a maintainer.

## Possible next steps

- Import all changes from `bitcoin::psbt` master branch.
- Improve the signing API.
- Refactor and clean it up if we get enthusiastic maintainers.
- Implement PSBTv2 (either from scratch or based on the released but incomplete `psbt-v2` crate) .
- Re-write the whole API using knowledge we learned over the last few years. 

See #3210 for context on crate creation.

**Comment by tcharding** [comment]
cc @DanGould @notmandatory

**Reply by apoelstra** [reply]
I can be a maintainer. Though I am also not an active user.

**Comment by pythcoiner** [comment]
i'm an active user, so if i can be of some help...

**Reply by tcharding** [reply]
Thanks for the offer, all and any help is most appreciated. I don't know who is behind this nym so all I can judge you buy is your contributions. I look forward to hacking with you!

**Comment by matthiasdebernardini** [comment]
I'm an active user working on a miniscript multisig coordinator, I would be happy to volunteer. However, I would need to know better what the time commitment is and what the expectations are.

**Reply by tcharding** [reply]
For readers, both @apoelstra and I have met @matthiasdebernardini in person at the recent Rust Bitcoin event in Nashville which adds some nice social glue to his offer.

**Comment by tcharding** [comment]
Thanks for the responses. I just realized that because this crate may/should end up being depended on by `miniscript`, which is @apoelstra's crate, and he is the BDFL in `rust-bitcoin` then it is really up to him to add maintainers, and I'm bang out of order even raising this discussion - my bad.

Now I've done it might as well see the discussion through, please take everything I say with a grain of salt.

The point of adding maintainers is to get buy in / commitment from them. There are however no fixed time requirements, its open source, just do what you can. I don't really know how to explain what is expected of a maintainer, its a personal burden you have to carry in a way you are comfortable with. But like I said I'm speaking out of line here.

**Reply by apoelstra** [reply]
I don't think you're out of order.

**Reply by tcharding** [reply]
Cool, cheers.

**Comment by tcharding** [comment]
@matthiasdebernardini perhaps follow the repo and just try act like a maintainer for a while and see if you like it - said another way just shoulder what you can and whatever you choose to take on do it well.

**Comment by ChrisCho-H** [comment]
I'd like to help if possible.

**Comment by tcharding** [comment]
Thanks! I'm not actually sure if this crate is going to get traction, I'm working on another one also. Maybe sit tight for a few weeks and see where we go.

**Comment by oliwiciaeth** [comment]
yeah thank for that information

**Comment by tcharding** [comment]
This crate is dead in the water. We are still trying to solve the PSBT problem: #5621
